tests/end_to_end/CairoPrecompiles/test_dual_vm_token.py

```diff
@@ -65,23 +66,50 @@ async def test_should_return_total_supply(self, starknet_token, dual_vm_token):
             total_supply_evm = await dual_vm_token.totalSupply()
             assert total_supply_starknet == total_supply_evm
 
-        async def test_should_return_balance_of(
-            self, starknet_token, dual_vm_token, owner
+        @pytest.mark.parametrize(
+            "signature,account_address",
+            [
+                ("balanceOf(address)", lambda account: account.address),
+                (
+                    "balanceOf(uint256)",
+                    lambda account: account.starknet_contract.address,
+                ),
+            ],
+        )
+        async def test_should_return_balance(
+            self, starknet_token, dual_vm_token, owner, signature, account_address
         ):
             (balance_owner_starknet,) = await starknet_token.functions[
                 "balance_of"
             ].call(owner.starknet_contract.address)
-            balance_owner_evm = await dual_vm_token.balanceOf(owner.address)
+            balance_owner_evm = await dual_vm_token.functions[signature](
+                account_address(owner)
+            )
             assert balance_owner_starknet == balance_owner_evm
 
+        async def test_should_revert_balance_of_invalid_address(
+            self, starknet_token, dual_vm_token
+        ):
+            evm_error = keccak("InvalidStarknetAddress()".encode())[:4]
+            with cairo_error(evm_error):
+                await dual_vm_token.functions["balanceOf(uint256)"](2**256 - 1)
```

Comment:
use `DEFAULT_PRIME` here instead

Reply:
Why ?
This is solidity and this could be passed as a valid value to the function